lib/go-tc/deliveryservices.go

```diff
@@ -560,6 +767,40 @@ func (ds *DeliveryServiceNullableV30) validateTypeFields(tx *sql.Tx) error {
 	return nil
 }
 
+func (ds *DeliveryServiceV40) Validate(tx *sql.Tx) error {
```

Comment:
Validate is used externally (DS Requests)

Reply:
Ah, I suppose it could be a public function in the `deliveryservice` package, which DSRs could import to use. That would still be better than it living in `lib/go-tc` IMO.

Reply:
There's a TODO somewhere in the codebase about moving validator methods internal to TO; I don't think this is the only handler that has this issue. It is really weird to be requiring a transaction hooked into a TO database in the supposedly generic library code, but moving it to be internal means re-thinking the ParseValidator interface and we were thinking of doing that anyway because of its current inability to return system-level errors. So it might be something to worry about in a later change.

Reply:
`TODeliveryService` doesn't implement the `ParseValidator` interface, so we are free to move this method into TO.